### ISSUE TITLE
Map preview/zoom option in the game lobby

### ISSUE THREAD

**Issue by pi99y** [OpenRA/OpenRA#7370]
In the game lobby it would be helpful (for strategic planning) to be able to see a higher resolution preview of the map especially when you're not familiar with it. The larger preview would also help with additional icon placement described in issue  #3359.

A click on the current map thumbnail would bring up a lager preview, another click (anywhere) or the Esc key would close it.

What do you think? 

Something like this:
![openra_lobby_map](https://cloud.githubusercontent.com/assets/10639257/5852543/e2f11be2-a218-11e4-963a-de8d9459079b.png)

![openra_lobby_map_preview](https://cloud.githubusercontent.com/assets/10639257/5852544/e7f87d7e-a218-11e4-81e1-8f85bbd2e289.png)

**Comment by penev92** [comment]
I've been thinking about this for a while.
We do need the option to generate full-sized image previews of the maps for several reasons, yours being one of them.
But I'm not familiar with our file packaging. If someone can give me a few pointers, I'd like to give this a try.

**Comment by penev92** [comment]
Update:
This is generated from the `pasty-mesa.oramap` file:
![](http://i.imgur.com/dzyj5SA.jpg)

And here are some for you RA lovers:
![](http://i.imgur.com/B51xRkh.png)
(east-vs-west)

![](http://i.imgur.com/RWZMnjZ.jpg)
(regeneration-basin)

![](http://i.imgur.com/oY1YSvj.jpg)
(seaside-2)


All these are generated directly from the `oramap` files ;)

**Comment by penev92** [comment]
I think the best way would be to have the map editor generate the preview and include it in the `oramap` file.
Then the lobby and resource site can both use that without having to generate it again and again, and also everyone can just open the file and see it without having to open the site or game.

**Comment by pi99y** [comment]
Looks and sounds good to me!
Maybe (if the map file size increase proves to be a problem) each client should generate and keep a cache of previews locally when the map is installed for the first time and then remembering a ~checksum of a map so it can recall that preview later... But I'm just guessing and will let the experts decide what's the best solution.

**Comment by Mailaender** [comment]
Depends on https://github.com/OpenRA/OpenRA/issues/2762.

**Comment by penev92** [comment]
I wouldn't say "depends". More like "does what was requested in that issue".

**Comment by Mailaender** [comment]
How did you do it? I assume you used the legacy OpenRA.Editor renderer we just dumped?

**Comment by penev92** [comment]
Yes. As that used to just draw the map tile by tile onto a bitmap, I simply made another copy of the bitmap and saved it as a file. That was questionable (and rejected) at the time, and impossible now that we dropped the old map editor.
